lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2230,6 +2258,47 @@ public function get_styles_for_block( $block_metadata ) {
 			}
 		}
 
+		// If there are style variations, generate the selectors for them, including any feature selectors.
+		$style_variation_declarations = array();
+		if ( ! empty( $block_metadata['variations'] ) ) {
+			foreach ( $block_metadata['variations'] as $style_variation ) {
+				$style_variation_node     = _wp_array_get( $this->theme_json, $style_variation['path'], array() );
+				$style_variation_selector = $style_variation['selector'];
+
+				if ( ! empty( $block_metadata['features'] ) ) {
```

Comment:
Would it be worth adding a comment prior to this line that this section processes feature level selectors for each style variation?

Reply:
It's always worth adding comments! Also I need to get better at remembering that 😅